cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+}
+
+void SpoilerBackgroundUpdaterThread::downloadSpoilerSeasonStatusFile()
+{
+    qDebug() << "Checking spoiler season status";
+    if (mpNetworkAccessManager == nullptr)
+    {
+        mpNetworkAccessManager = new QNetworkAccessManager(this);
+    }
+
+    QNetworkReply *lpNetworkReply = mpNetworkAccessManager->get(QNetworkRequest(QString(GITHUB_SPOILERS_STATUS_URL)));
+
+    // Force the calling of actSeeIfSpoilerSeasonIsActive() in sequence instead of parallel
+    // as we need to get that result before we start this timer thread up.
+    QEventLoop lcSubLoop;
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actSeeIfSpoilerSeasonIsActive()));
+    connect(lpNetworkReply, SIGNAL(finished()), &lcSubLoop, SLOT(quit()));
+    lcSubLoop.exec();
+}
+
+void SpoilerBackgroundUpdaterThread::actDownloadFinishedTokensFile()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QWidget *lpTempWidget = new QWidget();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerSavePath();
+
+    // Center the popup(s) that are to come in the near future
+    lpTempWidget->move(QApplication::desktop()->screen()->rect().center() - lpTempWidget->rect().center());
+    
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        QMessageBox::critical(lpTempWidget, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+    }
+    else
+    {
+        int lnStatusCode = lpNetworkReply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+        if (lnStatusCode == 301 || lnStatusCode == 302)
+        {
+            // Was a redirect, will need to start over!
+            QUrl lcRedirectUrl = lpNetworkReply->attribute(QNetworkRequest::RedirectionTargetAttribute).toUrl();
+            downloadSpoilersFile(lcRedirectUrl);
+        }
+        else
+        {
+            // Read downloaded file into buffer
+            mcDownloadedFileData = lpNetworkReply->readAll();
+
+            if (saveSpoilersToFile(lsSpoilerSavePath))
+            {
+                // Reload the card database to import the spoilers
+                QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+                QMessageBox::information(lpTempWidget, tr("Success"), tr("Spoiler database updated and saved successfully to\n%1").arg(lsSpoilerSavePath));
+            }
+            else
+            {
+                QMessageBox::critical(lpTempWidget, tr("Error"), tr("Spoiler database updated but could not be saved to %1").arg(lsSpoilerSavePath));
+            }
+        }
+}
+
+lpNetworkReply->deleteLater();
+lpTempWidget->deleteLater();
+}
+
+void SpoilerBackgroundUpdaterThread::actSeeIfSpoilerSeasonIsActive()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QWidget *lpTempWidget = new QWidget();
```

Comment:
Is this widget created only to pass a parent to QMessageBox? It should work the same if you pass 0 or nullptr.